### PR TITLE
Allow admin to update vacation request

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Allow admin to update vacation request

### Related Issues

- https://github.com/codescalers/CSHR/issues/397

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: client/src/components/cards/vacationCard.vue [hunk 1]
```diff
@@ -142,7 +144,7 @@ export default {
       if (user.value) {
         if (props.vacation.status == 'pending') {
           // could update if user signed in is the same user applied for vacation
-          if (props.vacation.isUpdated && user.value.fullUser.id == props.vacation.applying_user) {
+          if (props.vacation.isUpdated && user.value.fullUser.id == props.vacation.applying_user || user.value.fullUser.user_type != "User" ) {
```
Comment (mahmoud-emad) [inline client/src/components/cards/vacationCard.vue]: Why?

File: client/src/components/cards/vacationCard.vue [hunk 2]
```diff
@@ -66,6 +66,8 @@
       </v-form>
       <v-divider class="my-2"></v-divider>
       <v-row class="d-flex justify-end mt-3" v-if="couldApprove && vacation.status == 'pending'">
+        <v-btn color="primary" class="ma-1" type="submit"
+        :disabled="!form?.isValid" @click="updateVacation" v-if="vacation.applying_user.id !== user?.id">Update</v-btn>
```
Comment (mahmoud-emad) [inline client/src/components/cards/vacationCard.vue]: You should only allow admins who are working in the same office.

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

File: client/src/components/cards/vacationCard.vue (hunk 1 above)
Comment (zaelgohary) [reply client/src/components/cards/vacationCard.vue]: So any one with supervisor or admin rule could edit.

File: client/src/components/cards/vacationCard.vue (hunk 2 above)
Comment (zaelgohary) [reply client/src/components/cards/vacationCard.vue]: done.

File: client/src/components/cards/vacationCard.vue (hunk 1 above)
Comment (mahmoud-emad) [reply client/src/components/cards/vacationCard.vue]: The user also could update while the status of the request is pending

File: client/src/components/cards/vacationCard.vue [hunk 3]
```diff
@@ -66,6 +66,8 @@
       </v-form>
       <v-divider class="my-2"></v-divider>
       <v-row class="d-flex justify-end mt-3" v-if="couldApprove && vacation.status == 'pending'">
+        <v-btn color="primary" class="ma-1" type="submit"
+        :disabled="!form?.isValid" @click="updateVacation" v-if="vacation.applying_user.id !== user?.id && user?.fullUser.location.id == vacation.applying_user.location.id">Update</v-btn>
```
Comment (mahmoud-emad) [inline client/src/components/cards/vacationCard.vue]: Here are some cases you want to cover

1. if the requested user type is Admin: You need to make sure that the office is the same applier office.
2. if the requested user type is Supervisor: You need to check the approvals array and if the ID is included.
3. if the requested user type is User: You have to make sure that the applier ID is the same as the requested user ID.

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

File: client/src/components/cards/vacationCard.vue (hunk 3 above)
Comment (zaelgohary) [reply client/src/components/cards/vacationCard.vue]: > Here are some cases you want to cover
> 1. if the requested user type is Admin: You need to make sure that the office is the same applier office. 

Already applied.

> 2. if the requested user type is Supervisor: You need to check the approvals array and if the ID is included.

Working on it.

> 3. if the requested user type is User: You have to make sure that the applier ID is the same as the requested user ID. 

Already applied.

File: client/src/components/cards/vacationCard.vue (hunk 1 above)
Comment (zaelgohary) [reply client/src/components/cards/vacationCard.vue]: and that's what's already applied.

**Comment by Mahmoud-Emad** [comment]
please merge development

**Comment by Mahmoud-Emad** [comment]
Could be only one button to update
image

**Comment by Mahmoud-Emad** [comment]
The dates have not changed 

image
image

**Comment by zaelgohary** [comment]
> Could be only one button to update image

How to reproduce this?

**Comment by zaelgohary** [comment]
> The dates have not changed
> 
> image image

Already fixed [here](https://github.com/codescalers/CSHR/pull/461)

**Comment by Mahmoud-Emad** [comment]
> > The dates have not changed
> > image image
> 
> Already fixed [here](https://github.com/codescalers/CSHR/pull/461)

I think it shouldn't be added here

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**